app/src/main/java/com/example/benchproject/app/common/binding/recyclerView/RecyclerViewBindingAdapters.kt

```diff
@@ -2,14 +2,13 @@ package com.example.benchproject.app.common.binding.recyclerView
 
 import androidx.databinding.BindingAdapter
 import androidx.recyclerview.widget.RecyclerView
+import com.example.benchproject.app.common.recyclerView.BaseAdapter
 import com.example.benchproject.app.popular.movies.entity.Movie
-import com.example.benchproject.app.popular.movies.view.PopularMoviesRecyclerAdapter
 
 @BindingAdapter("updateAdapterData")
 fun RecyclerView.updateAdapter(data: List<Movie>) {
-    getRecyclerAdapter(this).uploadMovieData(data)
+    getRecyclerAdapter<BaseAdapter<Any>>().uploadData(data)
 }
 
-// TODO create generic solution for this
-private fun getRecyclerAdapter(recyclerView: RecyclerView): PopularMoviesRecyclerAdapter =
-    recyclerView.adapter as PopularMoviesRecyclerAdapter
+private fun <T> RecyclerView.getRecyclerAdapter() =
+    adapter as T
```

Comment:
This extension is not needed here you can move the code directly to `updateAdapter` function

Reply:
Ok, I changed it and now it looks like this: 
```
@BindingAdapter("updateAdapterData")
fun RecyclerView.updateAdapter(data: List<Movie>) {
    (adapter as BaseAdapter<Any>).uploadData(data)
}
```